tests/scripts/test_psa_compliance.py

```diff
@@ -49,12 +55,26 @@
 PSA_ARCH_TESTS_REPO = 'https://github.com/bensze01/psa-arch-tests.git'
 PSA_ARCH_TESTS_REF = 'fix-pr-5736'
 
-#pylint: disable=too-many-branches,too-many-statements
-def main():
-    mbedtls_dir = os.getcwd()
+#pylint: disable=too-many-branches,too-many-statements,too-many-locals
+def main(library_build_dir: str):
+    root_dir = os.getcwd()
+
+    in_psa_crypto_repo = build_tree.looks_like_psa_crypto_root(root_dir)
 
-    if not os.path.exists('library/libmbedcrypto.a'):
-        subprocess.check_call(['make', '-C', 'library', 'libmbedcrypto.a'])
+    if in_psa_crypto_repo:
+        crypto_lib_filename = \
+            library_build_dir + '/core/libpsacrypto.a'
+    else:
+        crypto_lib_filename = library_build_dir + '/library/libmbedcrypto.a'
+
+    if not os.path.exists(crypto_lib_filename):
+        #pylint: disable=bad-continuation
+        subprocess.check_call([
+            'cmake', '.',
+                     '-GUnix Makefiles',
+                     '-B', library_build_dir
+        ])
+        subprocess.check_call(['cmake', '--build', library_build_dir])
```

Comment:
```suggestion
        subprocess.check_call(['cmake', '--build', library_build_dir, '-t psacrypto/mbedcrypto'])
```
to build only the crypto library, not the other targets.